src/os/bluestore/BlueStore.cc

```diff
@@ -7909,6 +7937,14 @@ void BlueStore::_txc_update_store_statfs(TransContext *txc)
   txc->statfs_delta.encode(bl);
 
   txc->t->merge(PREFIX_STAT, "bluestore_statfs", bl);
+  string key;
+  OSDPool::get_key(txc->osd_pool_id, &key);
+  txc->t->merge(PREFIX_POOL_STAT, key, bl);
+  {
+    std::lock_guard<std::mutex> l(osd_pools_lock);
+    auto& p = osd_pools[txc->osd_pool_id];
+    p.stats += txc->statfs_delta;
+  }
```

Comment:
I wonder if this could *replace* the global counts and vstatfs_lock above.  if a caller asks for the global sum we can add it up (cheap,a nd not a frequent call).  And it means we're taking teh same number of locks and doing the same number of t->merge() calls as before (just spread over multiple keys instead of just one).

The hard part would be figuring out the upgrade path.  Not sure if/how that could be done.  hmm.